Headers/DebugServer2/Target/ThreadBase.h

```diff
@@ -57,6 +57,9 @@ class ThreadBase {
                            Address const &address = Address()) = 0;
 
 public:
+  virtual ErrorCode beforeResume();
```

Comment:
I don't think we'd even have to check if it's `kReasonThreadEntry`. Shouldn't breakpoints always be re-enabled when resuming?